src/container/builder.rs

```diff
@@ -0,0 +1,131 @@
+use crate::command::linux::LinuxSyscall;
+use std::path::PathBuf;
+
+use super::{init_builder::InitContainerBuilder, tenant_builder::TenantContainerBuilder};
+pub struct ContainerBuilder {
+    pub(super) container_id: String,
+
+    pub(super) root_path: PathBuf,
+
+    pub(super) syscall: LinuxSyscall,
+
+    pub(super) pid_file: Option<PathBuf>,
+
+    pub(super) console_socket: Option<PathBuf>,
+}
+
+/// Builder that can be used to configure the common properties of 
+/// either a init or a tenant container 
+/// 
+/// # Example
+///
+/// ```no_run
+/// use youki::container::builder::ContainerBuilder;
+///
+/// ContainerBuilder::new("74f1a4cb3801".to_owned())
+/// .with_root_path("/run/containers/youki")
+/// .with_pid_file("/var/run/docker.pid")
+/// .with_console_socket("/var/run/docker/sock.tty")
+/// .as_init("/var/run/docker/bundle")
+/// .build();
+/// ```
+impl ContainerBuilder {
+    /// Generates the base configuration for a container which can be
+    /// transformed into either a init container or a tenant container
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use youki::container::builder::ContainerBuilder;
+    ///
+    /// let builder = ContainerBuilder::new("74f1a4cb3801".to_owned());
+    /// ```
+    pub fn new(container_id: String) -> Self {
+        let root_path = PathBuf::from("/run/youki");
```

Comment:
It may be better to use Default Trait.

Reply:
Also, it's better to use Default for other builders as well.

Reply:
The default trait does not allow to specify arguments. I could do
```
ContainerBuilder::default()
.with_id(id)
```

instead of with new(id) and generate an id automatically if none is specified in with_id if you like. The TenantContainerBuilder and InitContainerBuilder are not intended to be instantiated directly but by calling as_tenant/is_tenant on the ContainerBuilder, so you do not really have a default value for them because you always have a builder that will be provided in new(...)

Reply:
I was planning to use default() in new, and I thought default would be better to use in case there are more constructor functions of different types.
However, it's a small thing, so I think it's fine to prepare it when it's needed.
What do you think?

Reply:
We only have one constructor function at the moment, so I think we should do a default implementation when we start to get multiple.